File: java/src/br/unifor/facilitador/porta/DocumentoUploadPort.java
package br.unifor.facilitador.porta;

import br.unifor.facilitador.dominio.Documento;
import java.util.UUID;

/** ISP: o app do paciente depende so disto. Nao enxerga metodo de leitura. */
public interface DocumentoUploadPort {
    UUID enviar(Documento d);
}

File: javascript/src/porta/DocumentoUploadPort.js
import { naoImplementado } from './_contrato.js';

// ISP: o app do paciente depende so disto. Nao enxerga metodo de leitura.
export class DocumentoUploadPort {
  enviar(documento) {
    naoImplementado(this.constructor.name, 'enviar');
  }
}
